### PR TITLE
remove dependency on sys-info crate

### LINKED ISSUE #9862: `Deno.osRelease()` in Windows
hi guys I think there is a problem extracting the OS version via the `Deno.osRelease()` method in Windows because when I run this code,
```ts
console.log(Deno.osRelease());
```
 I get `6.2.9200` in the console even though I am using Windows 10

Deno: v1.8.1
Windows: 10.0.19042

### PR DESCRIPTION
**Pull request by littledivy**
Fixes #9862 

`loadavg`

| Target family | Syscall | Description |
| ------------- | ------- | ----------- |
| Linux | `sysinfo` | - |
| Windows | - | Returns `DEFAULT_LOADAVG`. There is no concept of loadavg on Windows |
| macOS, BSD | `getloadavg` | https://www.freebsd.org/cgi/man.cgi?query=getloadavg |

`os_release`

| Target family | Syscall | Description |
| ------------- | ------- | ----------- |
| Linux | `/proc/sys/kernel/osrelease` | - |
| Windows | [`RtlGetVersion`](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-rtlgetversion) |  dwMajorVersion . dwMinorVersion . dwBuildNumber  |
| macOS | `sysctl([CTL_KERN, KERN_OSRELEASE])` | - |

`hostname`

| Target family | Syscall | Description |
| ------------- | ------- | ----------- |
| Unix | `gethostname(sysconf(_SC_HOST_NAME_MAX))` | - |
| Windows | `GetHostNameW` | - |

`mem_info`

| Target family | Syscall | Description |
| ------------- | ------- | ----------- |
| Linux | sysinfo | - |
| Windows | `sysinfoapi::GlobalMemoryStatusEx` | - |
| macOS | <br> <pre> sysctl([CTL_HW, HW_MEMSIZE]); <br> sysctl([CTL_VM, VM_SWAPUSAGE]); <br> host_statistics64(mach_host_self(), HOST_VM_INFO64) </pre> | - |

### CODE REVIEW

File: runtime/ops/os/sys_info.rs [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    let mut s = String::new();
+    File::open("/proc/sys/kernel/osrelease")?.read_to_string(&mut s)?;
+    s.pop(); // pop '\n'
+    s
+  }
+  #[cfg(target_vendor = "apple")]
+  {
+    let mut s = [0u8; 20];
+    let mut mib = [libc::CTL_KERN, libc::KERN_OSRELEASE];
+    let mut len = 20;
```
Comment (ry) [inline runtime/ops/os/sys_info.rs:58]: `s.len()` ?

is 20 enough?

File: runtime/ops/os/sys_info.rs [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+            "MemTotal" => mem_info.total = value,
+            "MemFree" => mem_info.free = value,
+            "MemAvailable" => mem_info.available = value,
+            "Buffers" => mem_info.buffers = value,
+            "Cached" => mem_info.cached = value,
+            "SwapTotal" => mem_info.swap_total = value,
+            "SwapFree" => mem_info.swap_free = value,
+            _ => (),
+          }
+        }
```
Comment (ry) [inline runtime/ops/os/sys_info.rs:188]: Can break the parser out into a separate function that operates on a string, and add a unit test on it?

File: runtime/ops/os/sys_info.rs [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+      .to_string()
+  }
+  #[cfg(target_family = "windows")]
+  {
+    use std::ffi::OsString;
+    use std::os::windows::ffi::OsStringExt;
+    use winapi::um::winsock2::GetHostNameW;
+
+    let namelen = 256;
+    let mut name: Vec<u16> = std::iter::repeat(0).take(namelen).collect();
```
Comment (ry) [inline runtime/ops/os/sys_info.rs:128]: `[0u16; namelen].to_vec()` ?

File: runtime/ops/os/sys_info.rs [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+type LoadAvg = (f64, f64, f64);
+const DEFAULT_LOADAVG: LoadAvg = (0.0, 0.0, 0.0);
+
+pub fn loadavg() -> LoadAvg {
+  #[cfg(target_os = "linux")]
+  {
+    fn read_loadavg() -> Result<String, std::io::Error> {
+      let mut s = String::new();
+      File::open("/proc/loadavg")?.read_to_string(&mut s)?;
```
Comment (billywhizz) [inline runtime/ops/os/sys_info.rs:11]: in general, using procfs is quite slow. there are alternatives we can use on modern linux for most of these places we read procfs. this also has benefit of not being dependent on procfs being mounted.

i raised an issue here on same theme: https://github.com/denoland/deploy/issues/3813

for loadavg, there is a sysinfo() syscall which gives us load average as well as all the memory stats. are there reasons we prefer to use procfs and/or cannot use sysinfo(), getrusage(), sysconf() etc. on linux?

https://man7.org/linux/man-pages/man2/sysinfo.2.html

Comment (billywhizz) [reply runtime/ops/os/sys_info.rs:11]: i did a little benchmark and pulling uptime etc. from sysyinfo syscall is ~2x doing an lseek/read on procfs for uptime alone (1.8m ops/sec v 800k). and it has the memory info we need in the same struct.

**Comment by billywhizz** [comment]
LGTM for the linux changes! :partying_face:

File: runtime/ops/os/sys_info.rs (hunk 1 above)
Comment (littledivy) [reply runtime/ops/os/sys_info.rs]: updated to 256

**Review by ry — APPROVED** [review]
LGTM

**Pull request merged**